web/packages/api/src/toEthereum_v3.ts

```diff
@@ -0,0 +1,856 @@
+
+export { ValidationKind, signAndSend } from "./toEthereum_v2"
+
+export async function createTransfer(
+    parachain: ApiPromise,
+    registry: AssetRegistry,
+    sourceAccount: string,
+    beneficiaryAccount: string,
+    tokenAddress: string,
+    amount: bigint,
+    fee: DeliveryFee
+): Promise<Transfer> {
+    const { ethChainId, assetHubParaId } = registry
+
+    let sourceAccountHex = sourceAccount
+    if (!isHex(sourceAccountHex)) {
+        sourceAccountHex = u8aToHex(decodeAddress(sourceAccount))
+    }
+
+    const sourceParachainImpl = await paraImplementation(parachain)
+    const { tokenErcMetadata, sourceParachain, ahAssetMetadata, sourceAssetMetadata } =
+        resolveInputs(registry, tokenAddress, sourceParachainImpl.parachainId)
+
+    let messageId: string | undefined = await buildMessageId(
+        parachain,
+        sourceParachainImpl.parachainId,
+        sourceAccountHex,
+        tokenAddress,
+        beneficiaryAccount,
+        amount
+    )
+    let tx: SubmittableExtrinsic<"promise", ISubmittableResult>
+    if (sourceParachainImpl.parachainId === assetHubParaId) {
+        tx = createAssetHubTx(
+            parachain,
+            ethChainId,
+            sourceAccount,
+            beneficiaryAccount,
+            ahAssetMetadata,
+            amount,
+            fee.totalFeeInDot,
+            fee.ethereumExecutionFee!,
+            messageId
+        )
+    } else {
+        tx = createSourceParachainTx(
+            parachain,
+            ethChainId,
+            assetHubParaId,
+            sourceParachainImpl.parachainId,
+            sourceAccountHex,
+            beneficiaryAccount,
+            sourceAssetMetadata,
+            amount,
+            fee.localExecutionFeeDOT! + fee.returnToSenderDeliveryFeeDOT,
+            fee.totalFeeInDot,
+            fee.assetHubExecutionFeeDOT +
+                fee.snowbridgeDeliveryFeeDOT +
+                fee.bridgeHubDeliveryFeeDOT,
+            fee.ethereumExecutionFee!,
+            messageId
+        )
+    }
+
+    return {
+        input: {
+            registry,
+            sourceAccount,
+            beneficiaryAccount,
+            tokenAddress,
+            amount,
+            fee,
+        },
+        computed: {
+            sourceParaId: sourceParachainImpl.parachainId,
+            sourceAccountHex,
+            tokenErcMetadata,
+            sourceParachain,
+            ahAssetMetadata,
+            sourceAssetMetadata,
+            messageId,
+        },
+        tx,
+    }
+}
+
+export async function getDeliveryFee(
+    connections: { assetHub: ApiPromise; source: ApiPromise; ethereum: AbstractProvider },
+    parachain: number,
+    registry: AssetRegistry,
+    tokenAddress: string,
+    padPercentage?: bigint,
+    defaultFee?: bigint
+): Promise<DeliveryFee> {
+    const { assetHub, source, ethereum } = connections
+    // Fees stored in 0x5fbc5c7ba58845ad1f1a9a7c5bc12fad
+    const feePadPercentage = padPercentage ?? 70n
```

Comment:
70% fee padding seems kinda of high, is there a reason for that?

Reply:
Reverted to 33% in https://github.com/Snowfork/snowbridge/pull/1492/commits/b3cefe9a68031e60119b985748b224274b67a0d4

I vaguely recall it's more of a workaround when I tested on local setup, but it does not really make sense. I'll double- check on Westend.